packages/components/src/segmented-control/component.tsx

```diff
@@ -74,29 +84,38 @@ function SegmentedControl(
 				'medium',
 				className
 			),
-		[ className ]
+		[ className, isBlock ]
 	);
 	return (
-		<SegmentedControlContext.Provider
-			value={ { ...radio, isBlock: ! isAdaptiveWidth } }
-		>
-			<RadioGroup
-				{ ...radio }
-				aria-label={ label }
-				as={ View }
-				className={ classes }
-				{ ...otherProps }
-				ref={ useMergeRefs( [ containerRef, forwardedRef ] ) }
+		<BaseControl aria-label={ label } help={ help } id={ id }>
+			<SegmentedControlContext.Provider
+				value={ { ...radio, isBlock: ! isAdaptiveWidth } }
 			>
-				{ resizeListener }
-				<Backdrop
+				{ ! hideLabelFromVision && (
+					<div>
+						<BaseControl.VisualLabel>
+							{ label }
+						</BaseControl.VisualLabel>
+					</div>
+				) }
```

Comment:
As [previously flagged](https://github.com/WordPress/gutenberg/pull/34017#discussion_r687006638), using `BaseControl.VisualLabel` seems the right approach here, given what explained in the [docs](https://developer.wordpress.org/block-editor/reference-guides/components/base-control/#basecontrol-visuallabel).

I just wonder if we should avoid passing `aria-label={ label }` to `<BaseControl />`

(cc @diegohaz in case you have any thoughts?)

Reply:
This makes sense. `id` is not required in code `BaseControl` and you can see this in source code here: https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/base-control/index.js#L65, but was in its definition. So I made it optional.

I updated to pass only `help`. Ideally we should pass the label there, but we have an `eslint` rule for when `label` is passed, an `id` should as well - that contradicts the implementation though..

commit: https://github.com/WordPress/gutenberg/pull/34017/commits/c718d7f9b4e30b7bcfcdc0671dd79b514eb2b417

Reply:
To be honest I don't understand why we need an "id" or "baseId" prop at all? I feel we should just remove them?